medusa/backup_manager.py

```diff
@@ -99,7 +99,7 @@ def set_backup_future(backup_name, future):
 
     # Sets the future for a backup; unknown on overall status at this point unless already existing
     @staticmethod
-    def register_backup(backup_name, is_async):
+    def register_backup(backup_name, is_async, overwrite_existing=True):
```

Comment:
question: In which case is the overwrite_existing=True and is_async=True ? I can't find such invocation in current codebase so I was wondering is the overwrite_existing=False intended to be used only with async cases?